mathesar_ui/src/stores/userProfile.ts

```diff
@@ -0,0 +1,76 @@
+import { setContext, getContext } from 'svelte';
+import { writable, type Readable, type Writable } from 'svelte/store';
+import UserApi, { type User, type UnsavedUser } from '@mathesar/api/users';
+
+const contextKey = Symbol('userprofile store');
+
+export class UserProfile implements Readonly<User> {
+  readonly id;
+
+  readonly is_superuser;
+
+  readonly database_roles;
+
+  readonly schema_roles;
+
+  readonly full_name;
+
+  readonly short_name;
+
+  readonly email;
+
+  readonly username;
+
+  constructor(userDetails: User) {
+    this.id = userDetails.id;
+    this.is_superuser = userDetails.is_superuser;
+    this.database_roles = userDetails.database_roles;
+    this.schema_roles = userDetails.schema_roles;
+    this.full_name = userDetails.full_name;
+    this.short_name = userDetails.short_name;
+    this.email = userDetails.email;
+    this.username = userDetails.username;
+  }
+
+  hasPermission() {
+    //
+  }
+
+  getDisplayName(): string {
+    if (this.short_name) {
+      return this.short_name;
+    }
+    return this.username;
+  }
+
+  /* @throws Error if unable to save */
+  async update(
```

Comment:
This feels fishy to me.

- We have classes like `ColumnsDataStore` which are very clearly **mutable** because:
    - Methods are named with verbs like `rename` and `patch`
    - Methods sometimes mutate the instance
    - Methods never return the instance
    - Methods sometimes mutate back-end data too

- We have classes like `Filtering` which are very clearly **immutable** because:
    - Methods are named with nouns and phrases like `withEntry` and `withoutColumns`
    - Methods never mutate the instance 
    - Methods always return a new instance
    - Methods never mutate backend data

- This `UserProfile` class doesn't fit squarely into either of the categories above, which makes me concerned that it may lead to confusion and readability problems. I would consider pulling the API logic out of the class, potentially into a new _mutable_ class called `UserProfileStore` which composes `UserProfile`.

Reply:
Thanks. Changes made in https://github.com/centerofci/mathesar/pull/2390/commits/031d821032982b2f174433e82ac3852a79b2e565

Reply:
This should also fix the `Function called outside component initialization` error.

Getting the context from out of a component initialization threw this error, I did a final set of refactoring while raising the PR and missed to factor this in.